packages/block-library/src/query/edit/inspector-controls/exclude-posts-control.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { FormTokenField } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { useDebounce } from '@wordpress/compose';
+import { useState, useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getEntitiesInfo } from '../../utils';
+import { MAX_SUGGESTED_POSTS } from '../../constants';
+
+/**
+ * Shared reference to an empty array for cases where it is important to avoid
+ * returning a new array reference on every invocation.
+ *
+ * @type {Array<any>}
+ */
+const EMPTY_ARRAY = [];
+
+const POSTS_QUERY = {
+	per_page: -1,
+	order: 'asc',
+	orderby: 'title',
+	_fields: 'id,title',
+};
+
+function ExcludePostsControl( { value, onChange, query } ) {
+	const [ search, setSearch ] = useState( '' );
+	const debouncedSearch = useDebounce( setSearch, 500 );
+	const searchResults = useSelect(
+		( select ) => {
+			const { getEntityRecords } = select( coreStore );
+			const _query = {
+				...POSTS_QUERY,
+				s: search,
```

Comment:
REST API's argument is `search` and not `s`. This results in fetching everything.
```suggestion
				search,
```